wgsl/index.bs

```diff
@@ -4561,6 +4561,54 @@ Example: `vec3(x,x,x)` is analyzed as follows:
 * The type of the expression is a [=vector=] of 3 components
     of [=i32=] (`vec3<i32>`).
 
+## Indeterminate values ## {#indeterminate-values}
+
+In limited cases, an evaluation of a [=runtime expression=] can occur
+using nonsense or unsupported values for its [=subexpressions=].
+
+In such a case, WGSL specifies that the result of that evaluation is
+an <dfn>indeterminate value</dfn> of the expression's [=static type=],
+meaning some arbitrary implementation-chosen value of the static type.
+
+Each distinct evaluation of the expression may yield a distinct value.
+For example, if the evaluation occurs once per iteration of a loop, a distinct
+value may be computed for each loop iteration.
+
+
+<div class='example wgsl global-scope' heading="Indeterminate value example">
+  <xmp highlight='rust'>
+    fn fun() {
+       var extracted_values: array<i32,2>;
+       const v = vec2<i32>(0,1);
+
+       for (var i: i32 = 0; i < 2 ; i++) {
+          // A runtime-expression used to index a vector, but outside the
+          // indexing bounds of the vector, produces an indeterminate value
+          // of the vector component type.
+          let extract = v[i+5];
+
+          // Now 'extract' is any value of type i32.
+
+          // Save it for later.
+          extracted_values[i] = extract;
+
+          if extract == extract {
```

Comment:
This holds because the type is i32.
For floats, the value could be NaN, and NaN is never equal to anything.
Expand this example to include NaN.

Reply:
Feedback from @jimblandy

Reply:
Addressed in the next commit.